src/plugins/action_generate_review_assignment_records/src/generateReviewAssignments.ts

```diff
@@ -1,112 +1,105 @@
-import { Reviewer, ReviewAssignmentObject, AssignmentStatus } from './types'
+import { Reviewer, ReviewAssignmentObject, AssignmentStatus, ApplicationData } from './types'
 import databaseMethods from './databaseMethods'
 
 module.exports['generateReviewAssignments'] = async function (input: any, DBConnect: any) {
   const db = databaseMethods(DBConnect)
 
   console.log('Generating review assignment records...')
   try {
-    const { applicationId, reviewId, templateId, stageId, stageNumber } = input
-    // NB: reviewId comes from record_id on TriggerPayload when triggered
-    // from review table
+    const { applicationId, reviewId } = input
+    // Get template information and current stage for application
+    const applicationData: ApplicationData = await DBConnect.getApplicationData(applicationId)
+    const { templateId, stageNumber, stageId } = applicationData
```

Comment:
I thought you were querying `stageNumber` using the amended `getCurrentReviewLevel` method (above)? This looks like you're still getting if from `applicationData`, which will be the stageNum that it was at at the start of this sequence of Actions -- which will cause the issue we discussed.

Reply:
Isn't the DBConnect going to get this from the DB instead of from the `applicationData` on send through the action?

And if that's is correct, I think I need to revert the change, since this Should NOT be the same as getting from `applicationData` received from the input

Reply:
If you want the current (i.e. new) stage details, just get it from `cumulativeOutput`, no need to make additional queries:
`const { applicationId, reviewId, applicationData, cumulativeOutput } = input`

Then you get templateId from applicationData and stageNumber and stageId from cumulativeOutput. Wiki currently describes what fields to expect from what Actions: https://github.com/openmsupply/application-manager-server/wiki/List-of-Action-plugins